wallet/walletrpc.c

```diff
@@ -198,13 +198,63 @@ static const struct json_command withdraw_command = {
 };
 AUTODATA(json_command, &withdraw_command);
 
+/* May return NULL if encoding error occurs. */
+static char *
+encode_pubkey_to_addr(const tal_t *ctx,
```

Comment:
This might actually be interesting to add to `pubkey.c` and make it globally available. In that case I'd rename it to `pubkey_to_addr` to match the rest of the functions there.

Reply:
Unfortunately it pulls in `common/bech32` due to `segwit_addr_encode` if we put it in `bitcoin/pubkey`.

The issue is that `common/test/Makefile` assumes that `bitcoin/*` modules will not depend on any `common/*` modules other than the trivial `common/utils` (which does not have its own test).  What happens is that every `common/test` now mocks `segwit_addr_encode`, and also need to include `common/bech32.h`.

`Makefile`s of each subdaemon also get affected (they now need to include `common/bech32`) even if subdaemons never need to print an onchain address anyway.

The effort does not seem commensurate to theoretical benefits...

Reply:
Good point, maybe we can reorg later a bit to move the bech32 dependency into `bitcoin/`, but that's for another time.